.github/workflows/_shared-first-time-setup.yml

```
@@ -0,0 +1,66 @@
+name: "Repository First Time Setup automation (shared)"
+
+on:
+  workflow_call:
+    inputs:
+        archetype_url:
+          type: string
+          default: 'https://changeme'
+          required: true
+        groupId:
+          type: string
+          default: ''
+          required: true
+        artifactId:
+          type: string
+          default: ''
+          required: true
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout ${{ github.ref }} branch in ${{ github.repository }} repository.
+      uses: actions/checkout@v3
+
+    - name: Download ESF archetype
+      if: startsWith(${{ github.event.inputs.archetype_url }}, 'https://eth-repo.s3.amazonaws.com/esf')
```

Looking at https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif we might want something like this:

```suggestion
  init:
    if: startsWith(github.event.inputs.archetype_url, 'https://eth-repo.s3.amazonaws.com/esf')
    runs-on: ubuntu-latest
    steps:

    - name: Checkout ${{ github.ref }} branch in ${{ github.repository }} repository.
      uses: actions/checkout@v3

    - name: Download ESF archetype
```